### DISCUSSION TITLE
Store own properties typing with Nuxt 3

### DISCUSSION THREAD

**Discussion by maxtomczyk** [vuejs/pinia#1241] [Help and Questions] [answered]
### Reproduction

no_reproduction

### Steps to reproduce the bug

1. Create new Nuxt 3 project with `nuxi init`.
2. Install pinia, @pinia/nuxt
3. Add `['@pinia/nuxt', { disableVuex: true }]` to `buildModules` in `nuxt.config.ts`
4. Add `@pinia/nuxt` to `compilerOptions.types` in `tsconfig.json`
5. Create some store

### Expected behavior

Store is able to resolve types of own properties. Example:
```typescript
import { defineStore } from 'pinia'

interface IState {
  a: string
}

export const useObrazkiStore = defineStore('test', {
  state: (): IState => ({
    a: 'someString'
  }),

  actions: {
    test() {
      const a = this.a // <- EXPECTED: this.a is of type string
      console.log(a)
    }
  }
})
```

### Actual behavior

Store is not able to resolve own types:
```typescript
import { defineStore } from 'pinia'

interface IState {
  a: string
}

export const useObrazkiStore = defineStore('test', {
  state: (): IState => ({
    a: 'someString'
  }),

  actions: {
    test() {
      const a = this.a // <- ACTUAL: this & this.a is of type any
      console.log(a)
    }
  }
})
```

### Additional information

Versions:
* nuxt 3: 3.0.0-27501460.04a72f8
* pinia: 2.0.13
* @pinia/nuxt: 0.1.8
* Typescript: 4.6.3

**Comment by posva** [comment]
Did you make sure you have `strict: true` in your tsconfig.json?

**Reply by Qendolin** [reply]
What if I don't want to use `strict: true` for my application? Is there another solution?

**Reply by runlong-yao** [reply]
it's working

**Comment by maxtomczyk** [comment, marked as answer]
Thanks for super fast response.

Looks like `strict: true` is partially answer for this question. Apparently nuxt 3 project created with `nuxi` doesn't support TS by default (types and configuration must be manually added).

**Reply by r0lodex** [reply]
Fast forward one year, how do you resolve this? I'm on nuxt 3.8 and pinia/nuxt 0.5.1, apparently the store still not able to resolve its own type
